src/encoding.rs

```diff
@@ -876,21 +878,16 @@ pub mod string {
 pub trait BytesAdapter: sealed::BytesAdapter {}
 
 mod sealed {
-    use super::{Buf, BufMut};
+    use super::Buf;
 
-    pub trait BytesAdapter: Default + Sized + 'static {
+    pub trait BytesAdapter: Default + Sized + AsRef<[u8]> + 'static {
```

Comment:
BytesAdapter is not used anymore for `encode`, but it seems to still be used for `encoded_len`. Will this mismatch cause issues for other people?